metrics/sender.go

```diff
@@ -50,8 +49,10 @@ type appInfo struct {
 	Version string `json:"version"`
 }
 
-type natMappingFailContext struct {
-	errorMessage string
+type natMappingContext struct {
+	stage        string
+	successful   bool
```

Comment:
Do we need to have bool `successful` here, or we can just check `if errorMessage != nil`?

Reply:
my guess is it makes the querying easier :)

Reply:
Well we can live without it now, but it is easier to query and understand events. Different NAT failures can have different metadata, i.e. if NAT hole-punching executes without errors, but  client connecting for VPN is unable to connect to us, we might create a failure event with more data, like `ip` of client, `port` which was used, protocol version and so on. So it might be difficult to see in a list of different fields that `error`/`errorMessage` is the key field to see whether it's event for successful or failed event.